boudicca.base/common-model/src/main/kotlin/base/boudicca/keyfilters/KeySelector.kt

```diff
@@ -105,6 +99,12 @@ class KeySelector private constructor(
             key.variants.forEach { builder.thenVariant(it) }
             return builder
         }
+
+        fun builder(keyFilter: KeyFilter): KeySelectorBuilder {
```

Comment:
while we're at it we could also extract this builder?

Also is there a reason why i would need to use the builder and cannot just pass a name and list to the constructor?

Reply:
yup, i will extract it. hm, not really. i think it makes it a bit more easier to reason about in this case, but i will make the constructor public, so you don't have to use the builder